monix-reactive/jvm/src/main/scala/monix/reactive/compress/internal/operators/GzipOperator.scala

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2014-2020 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.compress.internal.operators
+
+import java.nio.charset.StandardCharsets
+import java.time.Instant
+import java.util.zip.{CRC32, Deflater}
+
+import monix.execution.Ack
+import monix.execution.Ack.{Continue, Stop}
+import monix.reactive.Observable.Operator
+import monix.reactive.compress.internal.operators.Gzipper.gzipOperatingSystem
+import monix.reactive.compress.{
+  gzipCompressionMethod,
+  gzipExtraFlag,
+  gzipFlag,
+  gzipMagicFirstByte,
+  gzipMagicSecondByte,
+  zeroByte,
+  CompressionLevel,
+  CompressionParameters,
+  CompressionStrategy,
+  FlushMode
+}
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.Future
+
+private[compress] final class GzipOperator(
+  fileName: Option[String],
+  modificationTime: Option[Instant],
+  comment: Option[String],
+  bufferSize: Int,
+  params: CompressionParameters
+) extends Operator[Array[Byte], Array[Byte]] {
+  override def apply(out: Subscriber[Array[Byte]]): Subscriber[Array[Byte]] = {
+    new Subscriber[Array[Byte]] {
+      implicit val scheduler = out.scheduler
+
+      private[this] var isDone = false
+      private[this] var ack: Future[Ack] = _
+      private[this] val gzipper =
+        new Gzipper(
+          bufferSize,
+          params.level,
+          params.strategy,
+          params.flushMode,
+          fileName,
+          modificationTime,
+          comment
+        )
+
+      def onNext(elem: Array[Byte]): Future[Ack] = {
+        if (isDone) {
+          Stop
+        } else {
+          val result = gzipper.onChunk(elem)
+
+          // signaling downstream
+          ack = out.onNext(result)
+          ack
+        }
+      }
+
+      def onError(ex: Throwable): Unit =
+        if (!isDone) {
+          isDone = true
+          gzipper.close()
+          out.onError(ex)
+        }
+
+      def onComplete(): Unit =
+        if (!isDone) {
+          isDone = true
+          if (ack == null) ack = Continue
+          ack.syncOnContinue {
+            out.onNext(gzipper.finish())
```

Comment:
gzipper.finish() never throws, right?

Reply:
It seems that it can, should be fixed now.